indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -86,6 +88,11 @@
  */
 public class TaskLockbox
 {
+  private static final long LOCK_ACQUIRE_TIMEOUT_MILLIS = 10_000L;
+
+  private final Map<String, ReadWriteLock> datasourceToConcurrentLock = new HashMap<>();
```

Comment:
Even if locks don't take very long to be acquired, it is still not advisable to have a single lock per datasource. We need to include interval into the logic.

Reply:
One approach could be to maintain intervals currently held and waiting to be locked and implement a logic similar to shared and exclusive locks that already exists in TaskLockbox. When at transactional replace is released, any transactional append waiting on it can be signalled. However, it is always possible that one append is waiting on multiple replaces and vice versa.
I think that including the intervals into the logic makes it more involved and risky.

Could you please share any alternative suggestions to implement this?